packages/app/src/cli/models/extensions/extension-instance.ts

```diff
@@ -18,6 +18,52 @@ import {joinPath} from '@shopify/cli-kit/node/path'
 import {useThemebundling} from '@shopify/cli-kit/node/context/local'
 import {touchFile, writeFile} from '@shopify/cli-kit/node/fs'
 
+export class ConfigExtensionInstance<TConfiguration = unknown> {
```

Comment:
why create a new class? Isn't `ExtensionInstance` generic enough to support this?

Reply:
no, `ExtensionInstance` uses a different schema. The new `app_config` schema is generated dynamically from the `specs` schemas instead of having a fixed schema definition. Modifying the `ExtensionInstance` to support that required lots of changes in the code base. As this `App config` instances are quite simple, I think this is a better approach